lib/dal/src/action.rs

```diff
@@ -257,38 +253,23 @@ impl Action {
     ) -> ActionResult<Vec<Self>> {
         let workspace_snapshot = ctx.workspace_snapshot()?;
 
-        let nodes = workspace_snapshot
-            .outgoing_targets_for_edge_weight_kind(
+        let mut actions = vec![];
+        for (edge_weight, _, target_idx) in workspace_snapshot
+            .edges_directed_for_edge_weight_kind(
                 component_id,
-                EdgeWeightKindDiscriminants::Action,
+                Direction::Outgoing,
+                EdgeWeightKindDiscriminants::Func,
             )
-            .await?;
-        let mut node_weights = Vec::with_capacity(nodes.len());
-        let mut content_hashes = Vec::with_capacity(nodes.len());
-        for node in nodes {
-            let weight = workspace_snapshot.get_node_weight(node).await?;
-            content_hashes.push(weight.content_hash());
-            node_weights.push(weight);
-        }
-
-        let content_map: HashMap<ContentHash, ActionContent> = ctx
-            .layer_db()
-            .cas()
-            .try_read_many_as(&content_hashes)
-            .await?;
-
-        let mut actions = Vec::with_capacity(node_weights.len());
-        for node_weight in node_weights {
-            match content_map.get(&node_weight.content_hash()) {
-                Some(content) => {
-                    // NOTE(nick,jacob,zack): if we had a v2, then there would be migration logic here.
-                    let ActionContent::V1(inner) = content;
-
-                    actions.push(Self::assemble(node_weight.id().into(), inner.clone()));
+            .await?
+        {
+            if let (EdgeWeightKind::Func(kind), Some(node_weight)) = (
+                edge_weight.kind(),
+                workspace_snapshot.get_node_weight(target_idx).await.ok(),
+            ) {
+                if kind.to_owned() == FuncKind::Action {
```

Comment:
nit: I don't think we need to "own" the kind. We could probably just compare to `&FuncKind::Action`.